.travis.yml

```diff
@@ -41,3 +43,9 @@ script:
   - export PATH=$HOME/.cargo/bin:$PATH
   - make ci-travis
 
+after_success:
+  - python3 -m venv env --without-pip
+  - source env/bin/activate
+  - curl https://bootstrap.pypa.io/get-pip.py | python3
```

Comment:
Why do you have to install `pip` via `curl`? Shouldn't pip already be installed? If it isn't why not install it through `apt`?

Reply:
`pip3` is not installed by default in travis builds without `language: python` (we use `language: rust`), only `pip` is (for `python2`).
I install it using `curl` instead of `apt` to install it in a virtual environment (otherwise I believe `sudo` is required to install packages using pip, but we use `sudo: false`). Though in hindsight I maybe could have used `pip install --user []` if i set the path first - see https://github.com/travis-ci/travis-ci/issues/3563

Reply:
Ok, can we use the solution in the link you provided? It sounds like that will save build/install time

Reply:
Done!